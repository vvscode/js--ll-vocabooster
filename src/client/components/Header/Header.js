import React from 'react';
import { Link } from 'react-router-dom';
import './Header.less';

class Header extends React.Component {
  render() {
    return (
      <div className="Header__root">
        <div className="Header__container">
          <Link to="/">Header</Link>
        </div>
      </div>
    );
  }
}

export default Header;
