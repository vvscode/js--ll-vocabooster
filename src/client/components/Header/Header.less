@import '@client/styles/variables.css';

:root {
  --brand-color: #61dafb;
}

.Header {
  &__root {
    height: var(--header-height);
    box-shadow: 0 3px 5px 0 rgba(73, 73, 73, 0.31);
    background-color: var(--light-background);
    z-index: 2;
  }

  &__container {
    margin: 0 auto;
    padding: 10px 0;
    max-width: var(--max-content-width);
  }

  &__logo {
    margin-top: -10px;
  }
}
