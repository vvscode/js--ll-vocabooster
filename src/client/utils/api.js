const getFormdata = data => {
  const formData = new FormData();
  Object.keys(data).forEach(key => formData.append(key, data[key]));
  return formData;
};

export const getWords = text =>
  fetch('/api/vocagrabber', {
    method: 'POST',
    body: getFormdata({ text }),
  })
    .then(r => r.json())
    .then(({ data: list }) =>
      list.map((item, index) => ({ ...item, id: index })),
    );

export const addWordsToLingualeoVocabulary = words => Promise.resolve();
