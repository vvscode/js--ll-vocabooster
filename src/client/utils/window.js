export const { document } = window;

export const setTitle = title => {
  window.document.title = title;
};

export default window;
