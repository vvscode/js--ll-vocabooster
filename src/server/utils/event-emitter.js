import { EventEmitter } from 'events';

const customEventEmitter = new EventEmitter();

export default customEventEmitter;
