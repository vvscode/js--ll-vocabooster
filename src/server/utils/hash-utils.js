import hash from 'object-hash';

const hashAny = (value, keysToExclude = []) => {
  if (!value) return {};

  return {
    val: value,
    key: hash(value, {
      algorithm: 'md5',
      excludeKeys: key => keysToExclude.includes(key),
    }),
  };
};

const hashArray = arr => {
  const newArray = [];

  arr.forEach(item => {
    newArray.push(hashAny(item));
  });

  return newArray;
};

export { hashArray, hashAny };
