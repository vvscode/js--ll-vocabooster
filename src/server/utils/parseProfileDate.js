import dateformat from 'dateformat';

export default function parseProfileDate(dateString) {
  try {
    return dateformat(dateString, 'isoDate');
  } catch (error) {
    console.error(error);
    return null;
  }
}
