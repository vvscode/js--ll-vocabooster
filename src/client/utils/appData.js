import window from './window';

const data = (window && window.App) || {};

export const isLoggedIn = () =>
  !!data.user && Object.keys(data.user).length > 0;

export default data;
