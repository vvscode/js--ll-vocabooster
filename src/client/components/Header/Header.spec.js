import React from 'react';
import { shallow } from 'enzyme';
import * as appData from 'client/utils/appData';

import Header from './Header';

jest.mock('client/utils/appData');

describe('Header', () => {
  let props;
  let header;
  let appDataDefault;

  beforeEach(() => {
    props = {
      /* List all properties with 'undefined' values */
    };
    header = () => shallow(<Header {...props} />);
    appDataDefault = appData.default;
  });

  afterEach(() => {
    appData.default = appDataDefault;
  });

  it('renders correctly for authenticated user', () => {
    expect(header()).toMatchSnapshot();
  });
});
