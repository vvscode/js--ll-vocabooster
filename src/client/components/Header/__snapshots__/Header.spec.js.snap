// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly for authenticated user 1`] = `
<div
  className="Header__root"
>
  <div
    className="Header__container"
  >
    <Link
      replace={false}
      to="/"
    >
      Header
    </Link>
  </div>
</div>
`;
